compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECoreLoopUnroll.cpp

```diff
@@ -25,20 +25,17 @@ LogicalResult coreLoopUnroll(RewriterBase &rewriter, AMDAIE::CoreOp coreOp) {
     llvm::SmallDenseSet<unsigned> depths;
     for (auto acqOp :
          forOp.getBody()->getOps<AMDAIE::LogicalObjectFifoAcquire>()) {
-      auto stridedOp = dyn_cast_if_present<DoublyStridedCopyOpInterface>(
-          acqOp.getDma().getDefiningOp());
-      if (!stridedOp) {
-        acqOp.emitOpError()
-            << "should operate on a doubly strided copy operation for core "
-               "loop unrolling to behave correctly";
+      auto copyOp = dyn_cast_if_present<CopyOpInterface>(acqOp.getDma().getDefiningOp());
```

Comment:
Why we didn't use `DoublyStridedCopyOpInterface` here?

Is it because we now have `AMDAIE_NpuCircularDmaCpyNdOp` which is not copy-like ? (The only difference between `DoublyStridedCopyOpInterface` and `DoublyStridedOpInterface` is the presence of `CopyOpInterface` in the former)

Reply:
Because the input is not a `CircularDmaCpyNdOp` anymore, but a `ConnectionOp`. Ìn general, this pass does not expect a strided op to be its input, just an op with a source and target (`CopyOpInterface`)